client/my-sites/plan-features-2023-grid/components/features.tsx

```diff
@@ -21,20 +21,17 @@ const SubdomainSuggestion = styled.div`
 	}
 `;
 
-const FreePlanCustomDomainFeature: React.FC< { paidDomainName: string } > = ( {
-	paidDomainName,
-} ) => {
-	const {
-		data: wordPressSubdomainSuggestions,
-		isInitialLoading,
-		isError,
-	} = DomainSuggestions.useGetWordPressSubdomain( paidDomainName );
-
+const FreePlanCustomDomainFeature: React.FC< {
+	paidDomainName: string;
+	wpcomFreeDomainSuggestion: SingleFreeDomainSuggestion;
+} > = ( { paidDomainName, wpcomFreeDomainSuggestion } ) => {
 	return (
 		<SubdomainSuggestion>
 			<div className="is-domain-name">{ paidDomainName }</div>
-			{ isInitialLoading && <LoadingPlaceHolder /> }
-			{ ! isError && <div>{ wordPressSubdomainSuggestions?.[ 0 ]?.domain_name }</div> }
+			{ wpcomFreeDomainSuggestion.isLoading && <LoadingPlaceHolder /> }
+			{ wpcomFreeDomainSuggestion.entry && (
+				<div>{ wpcomFreeDomainSuggestion.entry.domain_name }</div>
```

Comment:
I'm thinking we shouldn't be letting these API snake_cases slip through to the rest of the code. Sounds to me like a transformation is missing at some layer from previously (pack/unpack the data structure sent/received from API). 🤔 

If there's a way to work around this, then great. Otherwise ok.

Reply:
Yeah, it makes sense. That needs to be fixed from https://github.com/Automattic/wp-calypso/blob/a51d5469dd913342a517cf37a0731c694d8e1c20/packages/data-stores/src/domain-suggestions/resolvers.ts#L53 if I'm not mistaken. It also touches many other places using it, so let's address it as a follow-up when we have a chance :)